### PR TITLE
[trainer] fix: pass labels to model in DPO concatenated_forward

### PR DESCRIPTION
**Pull request by Winter-Dry**
Since #711 rewrote concatenated_forward to use the chunked log-probs kernel via return_log_probs=True, the model's fused-linear path is guarded by `if labels is not None:` — but _NON_MODEL_KEYS still stripped labels from model_inputs (a pre-#711 leftover to avoid the HF loss branch on the gather-on-logits path). Result: fused_linear_aux stays None and `outputs.fused_linear_aux.log_probs` raises AttributeError.

Empty the set so labels are forwarded; the chunk_logprobs kernel needs them for shift + ignore_index masking anyway.

### CODE REVIEW

**Review by Coach257 — APPROVED** [review]

File: veomni/trainer/text_dpo_trainer.py [hunk 1]
```diff
@@ -38,7 +38,7 @@
 
 logger = logging.get_logger(__name__)
 
-_NON_MODEL_KEYS = {"labels"}
+_NON_MODEL_KEYS = set()
```
Comment (FoolPlayer) [inline veomni/trainer/text_dpo_trainer.py:41]: Passing labels fixes return_log_probs, but under SP these labels are already globally shifted. For packed DPO segments, the chosen tail can target the rejected head. Please use segment-aware shifted labels or mask segment boundaries, and add an SP regression test.

**Review by FoolPlayer — APPROVED** [review]
